File: DSA/DP/knapsack.go
package main

import "fmt"

func knapsack(w int, profit []int, weight []int) {
	length := len(weight)
	dp := make([][]int, length+1)

	for i := range length {
		dp[i] = make([]int, length)
	}

	fmt.Println("dp", dp)
}

func main() {
	w := 4
	profit := []int{1, 2, 3}
	weight := []int{4, 5, 1}

	fmt.Println("hello world", w, profit, weight)
	knapsack(w, profit, weight)
}

File: DSA/DP/knapsack.py
from typing import List


def knapsack(w: int, profit: List[int], weight: List[int]) -> None:
    dp = [[0 for _ in range(w + 1)] for _ in range(len(weight))]
    print("dp", dp)


W, profit, weight = 4, [1, 2, 3], [4, 5, 1]
knapsack(W, profit, weight)

"""
from typing import List

def knapsack(w: int, profit: List[int], weight: List[int] ) -> None :
    dp =  [ [0 for _ in range(w+1)] for _ in range(len(weight))]
    print("dp",dp)
    
W,profit,weight = 4,[1, 2, 3],[4, 5, 1]
knapsack(W, profit, weight)
"""
